enhancements/ingress/optional-ingress.md

```diff
@@ -0,0 +1,226 @@
+
+#### Authentication operator
+
+No change to the authentication operator is needed until the implementation of the ingress capability for the standalone OpenShift.
+HyperShift doesn't use the authentication operator.
+
+#### Monitoring operator
+
+No change to the monitoring operator is needed until the implementation of the ingress capability for the standalone OpenShift.
+The monitoring operator skips the alert manager's route check if `ingress` cluster operator is not there.
+
+### Risks and Mitigations
+
+#### Partially implemented capability
+
+The ingress capability will become a part of the capability API. Meanwhile its implementation will be incomplete due to the missing support of the standalone OpenShift. However the usage of the capability is an important stepping stone, it unifies the implementation of the optional ingress for all the OpenShift installation types.
+
+#### Component routes
+
+In the absence of the ingress operator, the component routes cannot be customized by the owner components due to the lack of the RBAC.
+The cluster administrator can customize a component route with a custom TLS certificate secret but the component owner won't be able to access the secret. This may be misleading for the cluster administrators. However the component routes were never fully implemented on HyperShift (authentication operator is not used).
+
+### Drawbacks
+
+In the absence of the default ingress controller, the Route API is left unserved. The OpenShift components or workloads can create routes but no controller is available to back them up. This may be misleading for the users.
+
+## Design Details
+
+### Open Questions [optional]
+
+N/A
+
+### Test Plan
+TBD
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+N/A
+
+#### Tech Preview -> GA
+N/A
+
+#### Removing a deprecated feature
+N/A
+
+### Upgrade / Downgrade Strategy
+
+Same as the component selection [strategy](/enhancements/installer/component-selection.md#upgrade--downgrade-strategy).
+
+### Version Skew Strategy
+N/A
+
+### Operational Aspects of API Extensions
+N/A - no api extensions are being introduced in this proposal
+
+#### Failure Modes
+N/A - no api extensions are being introduced in this proposal
+
+#### Support Procedures
+N/A - no api extensions are being introduced in this proposal
+
+## Implementation History
+N/A
+
+## Alternatives
+
+### Implement the ingress capability for the standalone OpenShift
+
+The goal of this enhancement is to satisfy the HyperShift's requirements. The implementation of the ingress capability for the standalone OpenShift brings a lot of additional complexity, unnecessary for HyperShift. The installer and the authentication operator would need to be updated to tolerate the absence of the default ingress controller.
```

Comment:
The "lot of additional complexity" is only in the world where the installer doesn't block disabling this capability, right?

Reply:
Right. I didn't count the block of the ingress disabling as a the implementation for the standalone OpenShift. I thought of it as the guard which frees us from the implementation.